src/cpp/lpnamer/main/ProblemGeneration.cpp

```diff
@@ -44,30 +43,57 @@ void CreateDirectories(const std::filesystem::path& output_path) {
 }
 
 ProblemGeneration::ProblemGeneration(ProblemGenerationOptions& options)
-    : options_(options) {}
+    : options_(options) {
+  if (!options_.StudyPath().empty()) {
+    mode_ = Mode::ANTARES_API;
+  } else if (!options_.XpansionOutputDir().empty()) {
+    mode_ = Mode::FILE;
+  } else if (!options_.ArchivePath().empty()) {
+    mode_ = Mode::ARCHIVE;
+  }
+}
+
 std::filesystem::path ProblemGeneration::updateProblems() {
-  const auto xpansion_output_dir = options_.XpansionOutputDir();
+  using namespace std::string_literals;
+  std::filesystem::path xpansion_output_dir;
   const auto archive_path = options_.ArchivePath();
 
-  auto deduced_xpansion_output_dir =
-      options_.deduceXpansionDirIfEmpty(xpansion_output_dir, archive_path);
+  if (mode_ == Mode::ARCHIVE) {
+    xpansion_output_dir =
+        options_.deduceXpansionDirIfEmpty(xpansion_output_dir, archive_path);
+  }
+
+  if (mode_ == Mode::ANTARES_API) {
+    throw std::runtime_error("Mode API Not implemented yet");
+  }
+
+  if (mode_ == Mode::FILE) {
+    simulation_dir_ = options_.XpansionOutputDir();  // Legacy naming.
+    // options_.XpansionOutputDir() point in fact to a simulation output from
+    // antares
+  }
+
+  if (mode_ == Mode::ANTARES_API || mode_ == Mode::FILE) {
```

Comment:
Why `Mode::ANTARES_API` here as we would have already raised the not implemented error ? For the future maybe ?

Reply:
yes